### PR TITLE
Add anemo-cli crate for making service-specific anemo CLI tools

### PR DESCRIPTION
**Pull request by aschran**

### CODE REVIEW

File: crates/anemo-cli/src/call.rs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        .get(&service_name)
+        .expect("service is configured")
+        .method_map
+        .get(&method_name)
+        .expect("method is configured");
+    let result = (method_fn)(peer, request).await;
+    println!("{result}");
+
+    // Explicitly disconnect to avoid error:
+    // "ActivePeers should be empty after all connection handlers have terminated"
```
Comment (bmwill) [inline crates/anemo-cli/src/call.rs:31]: if we're seeing this error then there's probably a bug somewhere that needs to be fixed.

File: crates/anemo-cli/src/call.rs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+use anemo::Network;
+use anemo_tower::trace::TraceLayer;
+
+pub async fn run(
+    config: Config,
+    address: Box<str>,
+    server_name: String,
+    service_name: String,
+    method_name: String,
+    request: String,
```
Comment (bmwill) [inline crates/anemo-cli/src/call.rs]: nit: change all of these to `&str`

File: crates/anemo-cli/src/ping.rs [hunk 3]
```diff
@@ -0,0 +1,17 @@
+use crate::util;
+use anemo::Network;
+use anemo_tower::trace::TraceLayer;
+
+pub async fn run(address: Box<str>, server_name: String) {
```
Comment (bmwill) [inline crates/anemo-cli/src/ping.rs]: you could probably change these to `&str`

File: crates/anemo-cli/src/call.rs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+) {
+    let network = Network::bind("localhost:0")
```
Comment (bmwill) [inline crates/anemo-cli/src/call.rs]: binding on localhost will only work for anemo services running locally.

File: crates/anemo-cli/src/ping.rs [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+        .private_key(util::random_key())
+        .server_name(server_name)
+        .outbound_request_layer(TraceLayer::new_for_client_and_server_errors())
+        .start(tower::service_fn(util::noop_handle))
+        .unwrap();
+
+    match network.connect(address).await {
+        Ok(peer_id) => println!("successfully connected to peer with ID: {peer_id:?}"),
+        Err(e) => println!("error connecting: {e:?}"),
+    }
```
Comment (bmwill) [inline crates/anemo-cli/src/ping.rs]: Would it be possible to dedup this logic with the logic in call?

Comment (bmwill) [reply crates/anemo-cli/src/ping.rs]: (the creation of the local anemo endpoint and potentially the connecting to the peer)

File: crates/anemo-cli/src/call.rs (hunk 1 above)
Comment (aschran) [reply crates/anemo-cli/src/call.rs:31]: Probably, but I don't think it's in this PR?

Comment (bmwill) [reply crates/anemo-cli/src/call.rs:31]: no its not, just a comment that we should investigate

File: crates/anemo-cli/src/call.rs (hunk 2 above)
Comment (aschran) [reply crates/anemo-cli/src/call.rs]: Kept request as String cause it makes ownership within the future easier, others all `&str` now

File: crates/anemo-cli/src/call.rs (hunk 4 above)
Comment (aschran) [reply crates/anemo-cli/src/call.rs]: Thanks, changed to 0.0.0.0

File: crates/anemo-cli/src/ping.rs (hunk 5 above)
Comment (aschran) [reply crates/anemo-cli/src/ping.rs]: great idea, done

**Pull request merged**